docs/design/proposals/rbd-mirror.md

```diff
@@ -0,0 +1,73 @@
+# RBD MIRRORING
+
+RBD mirroring is a process of replication of RBD images between two or more
+Ceph clusters. Mirroring ensures point-in-time, crash-consistent RBD images
+between clusters, RBD mirroring is mainly used for disaster recovery (i.e.
+having a secondary site as a failover). See Upstream [Ceph
+documentation](https://docs.ceph.com/en/latest/rbd/rbd-mirroring) on RBD
+mirroring for complete information.
+
+## Architecture
+
+![mirror](rbd-mirror.png)
+
+## Design
+
+Currently, ceph-csi generates its unique ID for each RBD image and stores the
+corresponding PV name and the unique ID mapping, It creates the RBD image with
+the unique ID and returns the encoded value which contains all the required
+information for other operations. For mirroring, the same RBD image will be
+mirrored to the secondary cluster. As the journal(omap data) is not mirrored to
+the secondary cluster, The cephcsi cannot identify the RBD image corresponding
+to the PV wihtout omap data.
+
+**Note:** It's expected that the user/admin will create the static PVC's on the
+secondary site during the failover.
+
+**Note:** when the static PVC created on the secondary site we cannot use the
+volumeHandle to identify the omap data or the image anymore because as we have
+only PoolID and clusterID in the volumeHandle. we cannot identify the correct
```

Comment:
consider formatting `PoolID` (with **C**apital?), `clusterID` and `volumeHandle` as these are names in the implementation